sunpy/visualization/animator/line.py

```diff
@@ -95,10 +95,18 @@ def __init__(self, data, plot_axis_index=-1, axis_ranges=None, ylabel=None, xlab
             axis_ranges = None
         if axis_ranges is None or axis_ranges[self.plot_axis_index] is None:
             self.xdata = np.arange(data.shape[self.plot_axis_index])
+        # Else derive the xdata as the centers of the pixel/bin edges
+        # supplied by the user for the plotted axis.
         else:
-            # Else derive the xdata as the centers of the pixel/bin edges
-            # supplied by the user for the plotted axis.
-            self.xdata = edges_to_centers_nd(np.asarray(axis_ranges[self.plot_axis_index]),
+            # If the shape of the array is a 1D array, get the centers about axis=0
+            if np.asarray(axis_ranges[self.plot_axis_index]).ndim == 1:
+                self.xdata = edges_to_centers_nd(np.asarray(axis_ranges[self.plot_axis_index]), 0)
+
+            # Else calculate the centers across the plot_axis_index
+            # Note that we expect the axis_ranges[plot_axis_index] to be of same shape as of data
+            # i.e. axis_ranges[plot_axis_index].shape == data.shape
```

Comment:
If this is true, then we shouldn't have to calculate the centres...  I think you mean something we expect ```axis_ranges[plot_axis_index]``` to give the edges by ```xdata``` needs to be centres...

Reply:
Yeah, that seems fair, I meant the other part.

Reply:
OK.  Change the comment to reflect this.